contents/code/tilingmanager.js

```diff
@@ -575,6 +582,37 @@ function TilingManager(timer) {
                                       print(err, "in i3-layout-set-normal-mode");
                                   }
                               });
+        KWin.registerShortcut("TILING: Move Window To New Desktop",
+                              "Move Window To New Desktop",
+                              "Meta+Shift+D",
+                              function() {
+                                  try {
+                                      var client = workspace.activeClient;
+                                      if (client != null) {
+                                          var desktop = 0;
+                                          // find first empty desktop
+                                          for (var i = 1; i <= self.desktopCount; i++) {
+                                              if (self._isDesktopEmpty(i)) {
+                                                  desktop = i;
+                                                  break;
+                                              }
+                                          }
+                                          // if there is none, create a new one
+                                          if (desktop == 0 && self.desktopCount < 20) {
+                                              workspace.desktops += 1;
+                                              desktop = self.desktopCount;
+                                          }
+                                          // move the client and activate the desktop
+                                          if (desktop > 0) {
```

Comment:
Okay, so if there are 20 full desktops (20 being KWin's limit), this does nothing?

That's probably correct.

Reply:
Yes, it does nothing. Alas, I could not come with anything better.

Reply:
There's a few possibilities, but since this is an edge-case, I assume it would be quite confusing to have something happen.

Or do you reach 20 desktops in daily usage of this?

Reply:
Never happened to me so far. I'm usually capped at six.

Reply:
Yeah, so whatever happens here just needs to be non-awful. Which "nothing" is.